tools/downloader/src/open_model_zoo/model_tools/converter.py

```diff
@@ -104,103 +104,131 @@ def main():
 
-        expanded_mo_args = [
-            string.Template(arg).substitute(template_variables)
-            for arg in model.mo_args]
+            model_format = model.framework
 
-        for model_precision in sorted(model_precisions):
-            data_type = model_precision.split('-')[0]
-            mo_cmd = [str(args.python), '--', str(mo_path),
-                '--framework={}'.format(model_format),
-                '--data_type={}'.format(data_type),
-                '--output_dir={}'.format(output_dir / model.subdirectory / model_precision),
-                '--model_name={}'.format(model.name),
-                *expanded_mo_args, *extra_mo_args]
+            template_variables = {
+                'config_dir': _common.MODEL_ROOT / model.subdirectory,
+                'conv_dir': output_dir / model.subdirectory,
+                'dl_dir': args.download_dir / model.subdirectory,
+                'mo_dir': mo_path.parent,
+            }
 
-            reporter.print_section_heading('{}Converting {} to IR ({})',
-                '(DRY RUN) ' if args.dry_run else '', model.name, model_precision)
+            if model.conversion_to_onnx_args:
+                if not convert_to_onnx(reporter, model, output_dir, args, template_variables):
+                    telemetry.send_event('md', 'converter_failed_models', model.name)
+                    telemetry.send_event('md', 'converter_error', str({'error': 'convert_to_onnx-failed', 'model': model.name}))
+                    return False
+                model_format = 'onnx'
 
-            reporter.print('Conversion command: {}', _common.command_string(mo_cmd))
+            expanded_mo_args = [
+                string.Template(arg).substitute(template_variables)
+                for arg in model.mo_args]
 
-            if not args.dry_run:
-                reporter.print(flush=True)
+            for model_precision in sorted(model_precisions):
+                data_type = model_precision.split('-')[0]
+                mo_cmd = [str(args.python), '--', str(mo_path),
+                    '--framework={}'.format(model_format),
+                    '--data_type={}'.format(data_type),
+                    '--output_dir={}'.format(output_dir / model.subdirectory / model_precision),
+                    '--model_name={}'.format(model.name),
+                    *expanded_mo_args, *extra_mo_args]
 
-                if not reporter.job_context.subprocess(mo_cmd):
-                    return False
+                reporter.print_section_heading('{}Converting {} to IR ({})',
+                    '(DRY RUN) ' if args.dry_run else '', model.name, model_precision)
 
-            reporter.print()
+                reporter.print('Conversion command: {}', _common.command_string(mo_cmd))
 
-        return True
+                if not args.dry_run:
+                    reporter.print(flush=True)
+
+                    telemetry.send_event('md', 'converter_precision', model_precision)
```

Comment:
Not sure this makes sense as a separate event. Maybe it would instead be better to add the precision to the `mo-failed` error event so that we know which invocation of MO failed?

Reply:
moved after argument parsing